### PR TITLE
replace cx_Oracle to python-oracledb

### PR DESCRIPTION
**Pull request by snickerjp**
## What type of PR is this? 

- [x] Refactor
- [x] Feature
- [x] Other

## Description

This is a pull request to switch the Oracle Driver module from cx_Oracle to python-oracledb.

Please note the following points with this change.

- The supported Python version is 3.6 or higher.🐍
- The supported Oracle database version is 12c or higher.
  - Thin mode: 12c or higher.
  - Thick mode: 9.2 or higher. (with Oracle Client library)
- In Thin mode, you do not need the Oracle Client library⭕️
- The ENCODING option has been deprecated. Please set the NLS_LANG environment variable instead.❌ -> `The environment variable "NLS_LANG" is also used now.`
  - https://python-oracledb.readthedocs.io/en/latest/user_guide/globalization.html#setting-the-client-character-set
- For other considerations from cx_Oracle changes, please refer to the following documents.
  - https://oracle.github.io/python-cx_Oracle/#features
  - https://oracle.github.io/python-oracledb/#features
- This PR will not include changes to `requirements_all_ds.txt`.

Please review and let me know if you have any comments. Thank you.🙇‍♂️

## How is this tested?

- [x] Unit tests (pytest, jest)
- [x] E2E Tests (Cypress)
- [x] Manually

## Related Tickets & Documents

#6373
#6380

### CODE REVIEW

File: requirements_oracle_ds.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
-# Requires installation of, or similar versions of:
+# If you want to use thick mode, you need to install the Oracle client library.
+# reference. https://pypi.org/project/oracledb/#description
 #    oracle-instantclient12.2-basic_12.2.0.1.0-1_x86_64.rpm
 #    oracle-instantclient12.2-devel_12.2.0.1.0-1_x64_64.rpm
-cx_Oracle==5.3
+oracledb==1.3.2
```
Comment (justinclift) [inline requirements_oracle_ds.txt:5]: Looks like they released version 1.4.0 today too. :wink:

Comment (snickerjp) [reply requirements_oracle_ds.txt:5]: I will try to update to version 1.4.0.

**Comment by justinclift** [comment]
The error message being shown there felt very familiar to me, as I remember something very similar in the old "RedashCommunity" repository.

I *think* you can fix it by removing the `--record` option from `client/cypress/cypress.js`.

Like in this patch: https://github.com/RedashCommunity/redash/commit/a5d12b1740402a25dce6ded436af5e3fbd227d79

**Comment by snickerjp** [comment]
>I think you can fix it by removing the --record option from client/cypress/cypress.js.

Thank you very much.

I will try it.

**Comment by justinclift** [comment]
Awesome.  Looks like this `python-oracledb` concept works. :smile:

**Comment by snickerjp** [comment]
@justinclift 
When I create a pull request to getredash/redash, do I need to submit this commit with it? 3893072bfe4e09d564049ae28c4cf58c461b0821

**Comment by justinclift** [comment]
@snickerjp Nope.  That commit was just working around an issue in forked repositories.  The PR doesn't need it. :smile:

**Comment by snickerjp** [comment]
@justinclift 
I understand.
I will remove this commit when I submit a pull request to getredash/redash.

**Pull request closed without merging**